File: src/main/kotlin/me/alvr/asfui/views/Cards.kt
package me.alvr.asfui.views

import javafx.scene.Parent
import javafx.scene.control.Button
import me.alvr.asfui.ASFProcess
import me.alvr.asfui.Command
import tornadofx.Fragment
import tornadofx.action
import tornadofx.enableWhen

class Cards : Fragment() {
    override val root: Parent by fxml("/cards.fxml")

    private val cardsFarmButton: Button by fxid("cards_farm")
    private val cardsLootButton: Button by fxid("cards_loot")
    private val cardsLootAllButton: Button by fxid("cards_lootall")
    private val cardsUnpackButton: Button by fxid("cards_unpack")

    private val bot = params["bot"] as String
    
    init {
        cardsFarmButton.apply {
            action {
                runAsync {
                    Command.sendCommand(Command.generateCommand(Command.FARM, bot))
                }
            }
            enableWhen(ASFProcess.started)
        }

        cardsLootButton.apply {
            action {
                runAsync {
                    Command.sendCommand(Command.generateCommand(Command.LOOT, bot))
                }
            }
            enableWhen(ASFProcess.started)
        }

        cardsLootAllButton.apply {
            action {
                runAsync {
                    Command.sendCommand(Command.generateCommand(Command.LOOT_ALL, bot))
                }
            }
            enableWhen(ASFProcess.started)
        }

        cardsUnpackButton.apply {
            action {
                runAsync {
                    Command.sendCommand(Command.generateCommand(Command.UNPACK, bot))
                }
            }
            enableWhen(ASFProcess.started)
        }
    }
}

File: src/views/cards.py
import threading

from PyQt5 import uic
from PyQt5.QtWidgets import QWidget

from utils.command import generate_command, send_command, FARM, LOOT, LOOT_ALL, UNPACK
from utils.resources import resource_path


class Cards(QWidget):
    def __init__(self, bots):
        super().__init__()

        uic.loadUi(resource_path('src/resources/ui/cards.ui'), self)

        self.bots = bots
        self.cards_farm.clicked.connect(self.farm)
        self.cards_loot.clicked.connect(self.loot)
        self.cards_loot_all.clicked.connect(self.loot_all)
        self.cards_unpack.clicked.connect(self.unpack)

    def farm(self):
        command = generate_command(FARM, self.bots.currentText())
        threading.Thread(target=send_command, args=(command,), daemon=True).start()

    def loot(self):
        command = generate_command(LOOT, self.bots.currentText())
        threading.Thread(target=send_command, args=(command,), daemon=True).start()

    def loot_all(self):
        command = generate_command(LOOT_ALL, self.bots.currentText())
        threading.Thread(target=send_command, args=(command,), daemon=True).start()

    def unpack(self):
        command = generate_command(UNPACK, self.bots.currentText())
        threading.Thread(target=send_command, args=(command,), daemon=True).start()
